overlord/snapstate/handlers.go

```diff
@@ -1942,6 +1942,46 @@ func (m *SnapManager) doPreferAliases(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+// reRefreshSetup holds the necessary details to re-refresh snaps that need it
+type reRefreshSetup struct {
+	Snaps  []string `json:"snaps,omitempty"`
+	UserID int      `json:"user-id,omitempty"`
+	*Flags
+}
+
+var reRefreshUpdater = updateManyFromChange
+
+func (m *SnapManager) doReRefresh(t *state.Task, tomb *tomb.Tomb) error {
```

Comment:
this doesn't have the triggering logic we discussed

Reply:
I think it does now.